src/Microsoft.TemplateEngine.Cli/TemplateResolution/TemplateGroup.cs

```diff
@@ -65,12 +65,26 @@ internal IReadOnlyList<string> ShortNames
             {
                 if (HasSingleTemplate)
```

Comment:
nit: We could drop this `if`(downside would be always creating `shortNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase);`